src/pages/popout-chat/App.js

```diff
@@ -0,0 +1,128 @@
+import React, { useState, useRef, useEffect, useCallback } from 'react';
+import styles from './submitForm.module.css';
+import TokenButton from '../../navigation/TokenButton/TokenButton';
+import MenuButton from '../../navigation/MenuButton/MenuButton';
+import * as Callbridge from '@iotum/callbridge-js';
+import { useSelector } from 'react-redux';
+
+function App() {
+  const [token, setToken] = useState('');
+  const [hostId, setHostId] = useState('');
+  const [submitted, setSubmitted] = useState(false);
+  const [unreadMessages, setUnreadMessages] = useState(0);
+  const [isLoading, setIsLoading] = useState(true); //Loading status of unread messages
+  const [domain, setDomain] = useState('iotum.callbridge.rocks'); //default domain
+
+  const widget = useRef(null);
+
+  const handleTokenChange = (event) => {
+    setToken(event.target.value);
+  };
+  const handleHostIdChange = (event) => {
+    setHostId(event.target.value);
+  };
+
+  const handleDomainChange = (event) => {
+    setDomain(event.target.value);
+  };
+  
+  const handleSubmit = () => {
+  //on submit -> get the number of unread messages 
+    const container = document.createElement('div');
+    container.style.display = 'none';
+    const invisibleWidget = new Callbridge.Dashboard(
+      {
+        domain: 'iotum.callbridge.rocks',
+        sso: {
+          token: token,
+          hostId: hostId,
+        },
+        container: container,
+      },
+      "Team",
+    );
+
+    invisibleWidget.on('dashboard.UNREAD_MESSAGES', (data) => {
+    const sum = Object.values(data.rooms).reduce((m, n) => m + n, 0);
+    setUnreadMessages(sum);
+    setIsLoading(false);
+    });
+    // Hide the form page
+    setSubmitted(true);
+  };
+
+  const renderChatWidget = () => {
+    if (!widget.current || !widget.current.instance) {
+      console.log("Create a new widget because the window was closed/there never was a widget")
+        widget.current = new Callbridge.Dashboard(
+          {
+            domain: domain, // using the state variable for domain
+            sso: {
+              token: token,
+              hostId: hostId,
+            },
+            container: window,
+            target: {
+              name: "CallbridgeChatWidget",
+              checkExisting: true,
+            }
+          },
+          'Team',
+        );
+
+        widget.current.on('dashboard.NAVIGATE', (data)=> {
+          console.log("navigate event");
+        })
+    }
+      else if(widget.current.instance) {
```

Comment:
use code editor's built-in auto formatting function on all your code.

you could setup auto format when saving the file.